stan/math/prim/fun/interp1.hpp

```diff
@@ -1,45 +1,36 @@
 #ifndef STAN_MATH_PRIM_FUN_INTERP1_HPP
 #define STAN_MATH_PRIM_FUN_INTERP1_HPP
 
-#include <stan/math/prim/meta.hpp>
+//#include <stan/math/prim/fun/sort_asc.hpp>
+
 using namespace std;
 namespace stan {
     namespace math {
-    /**
-     * Return the one-dimensional interpolation of the second argument of points corresponding to the first argument.
-     * @tparam T_true type of the true argument
-     * @tparam T_false type of the false argument
-     * @param c Boolean condition value.
-     * @param y_true Value to return if condition is true.
-     * @param y_false Value to return if condition is false.
-     */
-    inline Eigen::VectorXd
-    interp1(const Eigen::VectorXd& xData, const Eigen::VectorXd& yData, const Eigen::VectorXd& xVector)
-    {
-
-        Eigen::VectorXd yVals;
-        int size_x = xData.size();
-        int size_xData = xData.size();
+        template <typename T>
+        inline std::vector<double> interp1(const std::vector<double>& xData, const std::vector<double>& yData, const std::vector<T>& xTest)
+        {
+            using stan::math::index_type_t;
```

Comment:
Since we're not making assumptions about the input we need to check those assumptions:

```
check_size_match("interp1", "x", xDataPre.size(), "y", yDataPre.size());
check_sorted("interp1", "x", xDataPre);
```

Reply:
done